diablo-1250 - Update admin view of user page

mohamalnadi opened the pull request:
Jira: https://jira-secure.berkeley.edu/browse/DIABLO-1250

mohamalnadi committed:
diablo-1250 - Update admin view of user page

johncrossman commented on src/components/course/CoursesDataTable.vue:
```diff
@@ -236,6 +237,9 @@ import Days from '@/components/util/Days.vue'
 
+
+type TogglePayload = { sectionId: string | number; optedIn: boolean }
+
```
@mohamalnadi - I believe we want to remove `ToggleOptIn` from `CoursesDataTable`. What's the use case here? Afaik, `ToggleOptIn` should only be used on the /course page. (I'm have work in progress on toggle component that leverages `stores/course` (Pinia) and I'm concerned about a conflict in purpose.)

mohamalnadi replied:
@johncrossman - ahh, I had assumed we are keeping the opt-in toggle in the admin view of the user page, and only removing it from the instructor view. But yes, this make sense. I'll update the PR shortly to remove it from here as well.

johncrossman commented on src/views/User.vue:
```diff
@@ -204,4 +260,49 @@ const summarize = courses => {
+    })
+    .catch(() => {
+      emailReceive.value = !user.value.doNotEmail
+      alertScreenReader('Failed to update email preference.')
```
@mohamalnadi - Does the catch block above silence fatal errors? Would the user be aware that the update failed?

mohamalnadi replied:
@johncrossman - It does not, the diablo error page still shows up!

johncrossman commented on src/views/User.vue:
```diff
@@ -204,4 +260,49 @@ const summarize = courses => {
+        user.value.doNotEmail = !prefs.doNotEmail
+        emailReceive.value = !prefs.doNotEmail
+      }
+      if (typeof prefs?.optInNewCourses === 'boolean') {
```
@mohamalnadi - Just curious, why the frequent use of `typeof`? For example, in the code above, if the server returns an unexpected data type then our front-code would ignore that data and carry on. I'm concerned that this might hide potential errors in server-side code. Perhaps we can use Typescript checks to guarantee – in this case – boolean var types.

mohamalnadi commented:
@johncrossman - I just pushed some new changes. I removed the toggle from the CoursesDataTable and removed the `typeof` checks. Thank you for your feedback!

johncrossman commented on src/components/course/CoursesDataTable.vue:
```diff
@@ -297,8 +290,18 @@ const sortBy = defineModel(
+  emit('toggle', p)
+  props.onToggleOptIn(p)
+}
+
```
Thanks @mohamalnadi .  Can the two functions above be removed, too?  Also, do we still need the `TogglePayload` type?

johncrossman commented on src/components/course/ToggleOptIn.vue:
```diff
@@ -1,8 +1,9 @@
+<!-- ToggleOptIn.vue -->
```
@mohamalnadi - Can we revert all changes to `ToggleOptIn` since you've removed its usage from other components in this PR?  (I have working changes to `ToggleOptIn` and the changes here would cause conflicts in my work.)

mohamalnadi commented:
@johncrossman - I just reverted all the changes in the `ToggleOptIn` and removed the unnecessary code in the `CoursesDataTable` and fixed a few linter issues.

johncrossman commented:
Thanks @mohamalnadi. Looks good! Waiting on Travis.
